src/kOS/Execution/Stack.cs

```diff
@@ -48,9 +52,19 @@ private void ThrowIfInvalid(object item)
         /// <returns></returns>
         private object ProcessItem(object item)
         {
-            if (item is float)
+            if (item is float) {
                 // promote floats to doubles
-                return Convert.ToDouble(item);
+                item = Convert.ToDouble (item);
+            }
+            if (item is double)
+            {
+                if (!Double.IsNaN((double)item) && Int32.MinValue < (double)item && (double)item < Int32.MaxValue)
+                {
+                    int intPart = Convert.ToInt32(item);
+                    if ((double)item == (double)intPart)
+                        item = intPart;
+                }
+            }
```

Comment:
This is part of the logic that says in kOS, int and double numbers are interchangeable.  Rather than having each and every place that expects a number to be an int do the check, just make it so that when a round double is pushed, it becomes an int.

64-bit IEEE doubles can store accurate integers up to 2^53 before you hit the first integer that cannot be stored correctly without roundoff.  So the strict equality check works fine until you have integers bigger than a 32-bit integer anyway.